runtime/compiler/optimizer/J9TransformUtil.cpp

```diff
@@ -606,8 +606,13 @@ static void *dereferenceStructPointerChain(void *baseStruct, TR::Node *baseNode,
                   {
                   // if stable array, check that we are loading from the beginning of an element that it's not Null
                   // size of the element was checked in verifyFieldAccess
-                  if (fieldAddress % symRef->getSymbol()->getSize() != 0 ||
-                      isNullValueAtAddress(comp, symRef->getSymbol()->getDataType(), fieldAddress, symRef->getSymbol()))
+                  TR::DataType type = symRef->getSymbol()->getDataType();
+                  int32_t width = TR::Symbol::convertTypeToSize(type);
+                  if (comp->useCompressedPointers() && type == TR::Address)
+                     width = TR::Compiler->om.sizeofReferenceField();
```

Comment:
This will work, but it might be a little clearer to use `sizeofReferenceField()` whenever `type == TR::Address` regardless of `useCompressedPointers()`. That way we can see that we'll always use the right size (i.e. `sizeofReferenceField()`) for address-typed loads without having to reason that `convertTypeToSize()` will give the same value whenever `!useCompressedPointers()`

I wouldn't say it necessarily has to change, but I thought I'd mention it in case you think this is an improvement you'd like to make

Reply:
You are right about the message. I was just too brief. I will modify. Thanks!